File: src/main/java/com/timepath/util/logging/LogAggregator.java
package com.timepath.util.logging;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.LinkedList;
import java.util.logging.Handler;
import java.util.logging.LogRecord;

public class LogAggregator extends Handler {

    private final Collection<Handler> handlers = new LinkedList<>();

    public void addHandler(Handler h) {
        handlers.add(h);
    }

    @Override
    public void publish(LogRecord record) {
        for (@NotNull Handler h : handlers) {
            h.publish(record);
        }
    }

    @Override
    public void flush() {
        for (@NotNull Handler h : handlers) {
            h.flush();
        }
    }

    @Override
    public void close() {
        for (@NotNull Handler h : handlers) {
            h.close();
        }
    }
}

File: src/main/kotlin/com/timepath/util/logging/LogAggregator.kt
package com.timepath.util.logging

import java.util.LinkedList
import java.util.logging.Handler
import java.util.logging.LogRecord

public class LogAggregator : Handler() {

    private val handlers = LinkedList<Handler>()

    public fun addHandler(h: Handler) {
        handlers.add(h)
    }

    override fun publish(record: LogRecord) {
        for (h in handlers) {
            h.publish(record)
        }
    }

    override fun flush() {
        for (h in handlers) {
            h.flush()
        }
    }

    override fun close() {
        for (h in handlers) {
            h.close()
        }
    }
}
